### PR TITLE
HDDS-5467. Handle potential InterruptedException in Ozone

### PR DESCRIPTION
**Pull request by sky76093016**
## What changes were proposed in this pull request?

As described in [HDDS-5434](https://github.com/apache/ozone/pull/2413).

## What is the link to the Apache JIRA

https://issues.apache.org/jira/projects/HDDS/issues/HDDS-5467

## How was this patch tested?

No need.

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/util/ShutdownHookManager.java [hunk 1]
```diff
@@ -134,9 +130,12 @@ int executeShutdown() {
         future.cancel(true);
         LOG.warn("ShutdownHook '" + entry.getHook().getClass().
             getSimpleName() + "' timeout, " + ex.toString(), ex);
-      } catch (Throwable ex) {
+      } catch (InterruptedException ex) {
         LOG.warn("ShutdownHook '" + entry.getHook().getClass().
             getSimpleName() + "' failed, " + ex.toString(), ex);
+        Thread.currentThread().interrupt();
+      } catch (ExecutionException ex) {
+        ex.printStackTrace();
```
Comment (jojochuang) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/util/ShutdownHookManager.java:138]: please use LOG to record the exception.

Comment (sky76093016) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/util/ShutdownHookManager.java:138]: Of course, I immediately modify it.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/util/ShutdownHookManager.java [hunk 2]
```diff
@@ -35,11 +35,7 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Future;
-import java.util.concurrent.TimeoutException;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.*;
```
Comment (jojochuang) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/util/ShutdownHookManager.java:38]: I'm sorry forgot this one: can you not use wilcard import? Meaning: explicitly list each of the classes imported.

Comment (sky76093016) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/util/ShutdownHookManager.java:38]: I will fix it.

**Review by jojochuang — APPROVED** [review]

**Comment by jojochuang** [comment]
Will merge after precommit passes.

**Pull request merged**